### PR TITLE
Allow using python-app without a git checkout

### PR DESCRIPTION
**Pull request by jamielennox**
You shouldn't need to check something out from git to use python-app.
It's just a well-known location for a venv that installs the necessary
dependencies. Make the git checkout bit optional.

Signed-off-by: Jamie Lennox <jamielennox@gmail.com>

### CODE REVIEW

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.com/check_github/BonnyCI/hoist/213/1489123788.89/bonnyci-run-check-multinode : FAILURE in 20s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.com/check_github/BonnyCI/hoist/213/1489124685.96/bonnyci-run-check-multinode : FAILURE in 5m 49s

**Comment by ConsoleCatzirl** [comment]
What's the use-case for this?

File: roles/python-app/tasks/main.yml
Comment (omgjlk) [inline roles/python-app/tasks/main.yml]: I'm not sure I like putting the when before the block, but it's not technically wrong.

File: roles/python-app/defaults/main.yml
Comment (omgjlk) [inline roles/python-app/defaults/main.yml]: Since this defaults to False, do you need to update some calls to this role that depend on this functionality?

**Review by omgjlk — DISMISSED** [review]
Existing calls to this role need to turn on the `python_app_git_repo` fact.

**Comment by omgjlk** [comment]
Use case is to support re-using the roll to install things into a (new) venv. Came out of the PR for the ARA role.

**Comment by rattboi** [comment]
Once this is approved, I'll refactor the ARA role to use it. I didn't realize this was WIP when ARA role was merged.

**Comment by rattboi** [comment]
recheck

File: roles/python-app/tasks/main.yml
Comment (jamielennox) [reply roles/python-app/tasks/main.yml]: it just looked funny after because it was the only thing there.

File: roles/python-app/defaults/main.yml
Comment (jamielennox) [reply roles/python-app/defaults/main.yml]: no, it should be fine because all the existing roles define this already.

**Comment by jamielennox** [comment]
recheck

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/check_github/BonnyCI/hoist/213/1490152812.68/bonnyci-run-check-multinode : FAILURE in 6m 02s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/check_github/BonnyCI/hoist/213/1490154567.15/bonnyci-run-check-multinode : FAILURE in 1m 14s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/check_github/BonnyCI/hoist/213/1490154878.05/bonnyci-run-check-multinode : FAILURE in 1m 14s

**Comment by jamielennox** [comment]
recheck

**Comment by jamielennox** [comment]
recheck

**Review by mlangbehn — COMMENTED** [review]
Looks good to me, but others should review.

**Review by omgjlk — APPROVED** [review]

**Pull request merged**
